Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -141,13 +141,15 @@ namespace Terrain
         {
             // Rebuild the draw packets when the material or shaders change.
             RebuildDrawPackets();
+            m_candidateSectors.clear();
             m_rebuildDrawPackets = false;
         }
 
         if (m_rebuildSectors)
         {
             // Rebuild the sectors when the configuration or terrain world changes
             RebuildSectors();
+            m_candidateSectors.clear();
```

Comment:
RebuildSectors already clears m_candidateSectors.
I am thinking if rebuild and update operation trigger m_candidateSectors to update every time, should we just put m_candidateSectors.clear() inside those functions instead of having them paired outside?